packages/ss58_codec/test/ss58_codec_test.dart

```diff
@@ -56,78 +56,178 @@ void main() {
       expect(
-          () =>
-              SS58Codec.decode('fRWKeM1KzddF4G6N6isvg6SpFVWJLLXRyYvK1dXLx4xjP'),
-          throwsException);
+        () => SS58Codec.decode('fRWKeM1KzddF4G6N6isvg6SpFVWJLLXRyYvK1dXLx4xjP'),
+        throwsA(
+          predicate((exception) =>
+              exception is InvalidPrefixException &&
+              exception.toString() == 'Invalid SS58 prefix byte.'),
+        ),
+      );
     });
+
+    /// example of invalid decoded address length
     test('(data.length - offset) != any of [2, 3, 5, 9, 34, 35]', () {
       expect(
-          () => SS58Codec.decode(
-              '3HX1zEyzCbxeXe34JY8SNSVAZ6djFccsV5f67PTied4CcWHspQ'),
-          throwsException);
+        () => SS58Codec.decode(
+            '3HX1zEyzCbxeXe34JY8SNSVAZ6djFccsV5f67PTied4CcWHspQ'),
+        throwsA(
+          predicate((exception) =>
+              exception is BadAddressLengthException &&
+              exception.toString() == 'Bad Length Address.'),
+        ),
+      );
+      expect(
+        () => SS58Codec.decode(
+            '4pa95kBXvMqgbpDXkFVHE9JfnNqamjYQYtmSTZGcnBXwvnmosP'),
+        throwsA(
+          predicate((exception) =>
+              exception is BadAddressLengthException &&
+              exception.toString() == 'Bad Length Address.'),
+        ),
+      );
       expect(
-          () => SS58Codec.decode(
-              '4pa95kBXvMqgbpDXkFVHE9JfnNqamjYQYtmSTZGcnBXwvnmosP'),
-          throwsException);
+        () => SS58Codec.decode(
+            '3rr8zEMfiR2AjQYpVvncJCAcRP7CLzfxMGuUU2Pw2MCotrsQnS'),
+        throwsA(
+          predicate((exception) =>
+              exception is BadAddressLengthException &&
+              exception.toString() == 'Bad Length Address.'),
+        ),
+      );
       expect(
-          () => SS58Codec.decode(
-              '3rr8zEMfiR2AjQYpVvncJCAcRP7CLzfxMGuUU2Pw2MCotrsQnS'),
-          throwsException);
+        () => SS58Codec.decode(
+            '4uqUGom7PszVSaZipgHYwVNsMESj1W6cmMZYeXGFeeXGX6VEqm'),
+        throwsA(
+          predicate((exception) =>
+              exception is BadAddressLengthException &&
+              exception.toString() == 'Bad Length Address.'),
+        ),
+      );
+    });
+
+    /// example of invalid decoded address.
+    test('invalid check sum exception', () {
       expect(
-          () => SS58Codec.decode(
-              '4uqUGom7PszVSaZipgHYwVNsMESj1W6cmMZYeXGFeeXGX6VEqm'),
-          throwsException);
+        () => SS58Codec.decode(
+            '3HX1zEyzCbxeXe34JY8SNSVAZ6djFccsV5f67PTied4CcWHs'),
+        throwsA(
+          predicate((exception) =>
+              exception is InvalidCheckSumException &&
+              exception.toString() == 'Invalid checksum'),
+        ),
+      );
     });
   });
 
-  // Encode Exception testing
+  /// [SS58Codec] `encode` method Exception testing
   group('encode: throw exception -> ', () {
+    /// [Address] with invalid `prefix`
     test('invalid prefix ', () {
```

Comment:
Using BDD's naming convention will also clean up some of your comments. They are not bad, but are usually a code smell. Your code should read itself, and if you're having to explain it using comments, usually it's not readable.
Also, use `///` only for documentation. You can use common `//` for comments
```suggestion
    test('it should throw InvalidPrefixException when an address has invalid prefix', () {
```